File: Java/Leetcode_704_search.java
public class Leetcode_704_search {
    public int search(int[] nums, int target) {
        int left = 0;
        int right = nums.length - 1;
        while (left <= right ){
            int middle = (left+right)/2;
            if(nums[right] > target){
                right = middle;
            }
            else if (nums[left] > target){
                
            }
        }
        return target;
    }
    public static void main(){

    }
}

File: Python/Leetcode_704_search.py
def search(nums ,target):
    left = 0
    right = len(nums) - 1 
    while(left <= right):
        middle = int((left + right)/2)
        if (target < nums[middle]):
            right = middle-1
        elif(target > nums[middle]):
            left = middle +  1
        else:
            return middle
    return -1

nums = [-1,0,3,5,9,12]
target = 9
print(search(nums ,target))